turnerlabs/yt_push
# DRACO-3732 Support multiple media types

Adds support for multiple media types, avoiding issues with re-claiming auth codes, and making sure that the refresh token fetched against a media type is saved to other media types with the same client id.

**To test:**
- First ensure that your able to authenticate a single media type as before.
- Add a second media type with the same client id, channel and owner id. Ensure your able to authenticate it successfully, and that the refresh token was re-saved for both media types. The goal here is that refresh tokens should match across any media types with the same client id.
- Add a third media type with a different client id, channel and owner id. Ensure your able to authenticate and that the refresh token saved is only written to this new media type, and not the older two media types.

To get a second client id, follow these steps to get credentials for your own personal YouTube account and channel: https://developers.google.com/youtube/registering_an_application. Let me know if you have questions about how to set this up.
